tiledb/ml/data_apis/pytorch.py

```diff
@@ -0,0 +1,63 @@
+"""Functionality for loading data directly from dense TileDB arrays into the PyTorch Dataloader API."""
+import tiledb
+import torch
+import math
+
+
+class PyTorchTileDBDenseDataset(torch.utils.data.IterableDataset):
+    """
+    Class that implements all functionality needed to load data from TileDB directly to the
+    PyTorch Dataloader API.
+    """
+
+    def __init__(self, x_array: tiledb.Array, y_array: tiledb.Array, batch_size: int):
```

Comment:
Is there a reason that for Pytorch we initialise the object? Do we want that initialisation to happen during object creation? Or should we follow a similar approach to the TF readers with class initialisation? Would it make sense to follow a common front across integrations?

Reply:
It's a different case than Tensorflow here. In the Tensorflow case we had to return and object of type (from_generator) other than the class (tf.data.Dataset) it was  bound to, which couldn't happen without the use of __new__. With PyTorch we don't have to do this as we finally need an object of type torch.utils.data.IterableDataset.